File: scripts/test_solutions/253_meeting_rooms_ii.js
var minMeetingRooms = function(intervals) {
    const starts = intervals.map(i => i[0]).sort((a, b) => a - b);
    const ends = intervals.map(i => i[1]).sort((a, b) => a - b);
    let rooms = 0, endPtr = 0;
    for (let i = 0; i < starts.length; i++) {
        if (starts[i] < ends[endPtr]) rooms++;
        else endPtr++;
    }
    return rooms;
};

File: scripts/test_solutions/253_meeting_rooms_ii.cpp
class Solution {
public:
    int minMeetingRooms(vector<vector<int>>& intervals) {
        vector<int> starts, ends;
        for (const auto& i : intervals) {
            starts.push_back(i[0]);
            ends.push_back(i[1]);
        }

        sort(starts.begin(), starts.end());
        sort(ends.begin(), ends.end());

        int rooms = 0, end_ptr = 0;
        for (int i = 0; i < starts.size(); i++) {
            if (starts[i] < ends[end_ptr]) {
                rooms++;
            } else {
                end_ptr++;
            }
        }
        return rooms;
    }
};
